plugins/ingest/src/test/java/org/elasticsearch/ingest/IngestClientIT.java

```diff
@@ -55,6 +59,58 @@
 
     }
 
+    public void testSimulate() throws Exception {
+        new PutPipelineRequestBuilder(client(), PutPipelineAction.INSTANCE)
+                .setId("_id")
+                .setSource(jsonBuilder().startObject()
+                        .field("description", "my_pipeline")
+                        .startArray("processors")
+                        .startObject()
+                        .startObject("grok")
+                        .field("field", "field1")
+                        .field("pattern", "%{NUMBER:val:float} %{NUMBER:status:int} <%{WORD:msg}>")
+                        .endObject()
+                        .endObject()
+                        .endArray()
+                        .endObject().bytes())
+                .get();
+        assertBusy(new Runnable() {
+            @Override
+            public void run() {
+                GetPipelineResponse response = new GetPipelineRequestBuilder(client(), GetPipelineAction.INSTANCE)
+                        .setIds("_id")
+                        .get();
+                assertThat(response.isFound(), is(true));
+                assertThat(response.pipelines().get("_id"), notNullValue());
+            }
+        });
+
+        SimulatePipelineResponse response = new SimulatePipelineRequestBuilder(client(), SimulatePipelineAction.INSTANCE)
+                .setId("_id")
+                .setSource(jsonBuilder().startObject()
+                        .startArray("docs")
```

Comment:
this is also why I would like to move the parsing to the rest layer. java api users must build a  json at the moment, that's not the best... again we can work on this later though